src/safeposix/syscalls/sys_calls.rs

```diff
@@ -271,6 +298,104 @@ impl Cage {
         status
     }
 
+    pub fn waitpid_syscall(&self, cageid: i32, status: &mut i32, options: i32) -> i32 {
```

Comment:
Agree with what Justin said. Might be good to add a comment to explain high-level implementation logic here